### PR TITLE
fix(lidar_apollo_instance_segmentation): fix shadowVariable

### PR DESCRIPTION
**Pull request by kobayu858**
## Description
This is a fix based on cppcheck shadowVariable warnings

```
perception/lidar_apollo_instance_segmentation/src/debugger.cpp:87:17: style: Local variable 'i' shadows outer variable [shadowVariable]
    for (size_t i = 0; i < object_pointcloud.size(); i++) {
                ^
```
Using range-based for loop.

## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/lidar_apollo_instance_segmentation/src/debugger.cpp [hunk 1]
```diff
@@ -34,12 +34,12 @@ void Debugger::publishColoredPointCloud(
 {
   using autoware_perception_msgs::msg::ObjectClassification;
   pcl::PointCloud<pcl::PointXYZRGB> colored_pointcloud;
-  for (size_t i = 0; i < input.feature_objects.size(); i++) {
+  for (const auto & feature_object : input.feature_objects) {
```
Comment (veqcc) [inline perception/lidar_apollo_instance_segmentation/src/debugger.cpp:37]: @kobayu858 
Consider using range-based for loops

Comment (kobayu858) [reply perception/lidar_apollo_instance_segmentation/src/debugger.cpp:37]: @veqcc 
What do you mean by what I think is already a range-based for loop?

File: perception/lidar_apollo_instance_segmentation/src/detector.cpp [hunk 2]
```diff
@@ -182,9 +182,9 @@ bool LidarApolloInstanceSegmentation::detectDynamicObjects(
 
   // move down pointcloud z_offset in z axis
   for (std::size_t i = 0; i < output.feature_objects.size(); i++) {
-    sensor_msgs::msg::PointCloud2 transformed_cloud;
-    transformCloud(output.feature_objects.at(i).feature.cluster, transformed_cloud, -z_offset_);
-    output.feature_objects.at(i).feature.cluster = transformed_cloud;
+    sensor_msgs::msg::PointCloud2 updated_cloud;
+    transformCloud(output.feature_objects.at(i).feature.cluster, updated_cloud, -z_offset_);
+    output.feature_objects.at(i).feature.cluster = updated_cloud;
   }
```
Comment (veqcc) [inline perception/lidar_apollo_instance_segmentation/src/detector.cpp:188]: @kobayu858 
Could you use range-based for loop here too?

File: perception/lidar_apollo_instance_segmentation/src/debugger.cpp (hunk 1 above)
Comment (veqcc) [reply perception/lidar_apollo_instance_segmentation/src/debugger.cpp:37]: Sorry my comment was pointing to wrong line number.
I indicated the correct one in another comment.

File: perception/lidar_apollo_instance_segmentation/src/detector.cpp (hunk 2 above)
Comment (kobayu858) [reply perception/lidar_apollo_instance_segmentation/src/detector.cpp]: @veqcc 
Correction. In the future, if I find similar for loop even if it is not directly related to the cppcheck warning, should I fix it?

**Review by veqcc — APPROVED** [review]

File: perception/lidar_apollo_instance_segmentation/src/detector.cpp (hunk 2 above)
Comment (veqcc) [reply perception/lidar_apollo_instance_segmentation/src/detector.cpp]: If it is not a complicated fix, yes please.

Comment (kobayu858) [reply perception/lidar_apollo_instance_segmentation/src/detector.cpp]: I understand.

**Comment by kobayu858** [comment]
@YoshiRi @yukkysaito 

Please confirm the PR. Thank you very much.

**Review by yukkysaito — COMMENTED** [review]
What kind of tests were performed for this PR? I don't think there are any major issues, but I would like to keep a record of the tests that were conducted in advance.

**Comment by veqcc** [comment]
> What kind of tests were performed for this PR? I don't think there are any major issues, but I would like to keep a record of the tests that were conducted in advance.

When fixing `shadowVariable`, I'm telling that we need no to any tests, since it usually just changes variabled names.
If you want us conduct some tests, like Psim or Lsim, we are happy to do them. But I'm not sure they are really necessary for these kinds of simple modifications.

**Review by YoshiRi — APPROVED** [review]
@yukkysaito 
This PR won't change any actual behavior, so I think we can approve this.

**Pull request merged**